File: bots/stub-ts/StubBot.ts
import { Bot, HitWallEvent, ScannedBotEvent } from "@robocode.dev/tank-royale-bot-api";

class StubBot extends Bot {
  static main() {
    new StubBot().start();
  }

  override run() {
    while (this.isRunning()) {
      this.turnRadarRight(360);
      this.forward(40);
      this.turnLeft(15);
    }
  }

  override onScannedBot(_e: ScannedBotEvent) {
    this.fire(1);
  }

  override onHitWall(_e: HitWallEvent) {
    this.back(60);
    this.turnRight(90);
  }
}

StubBot.main();

File: bots/stub-java/StubBot.java
import dev.robocode.tankroyale.botapi.*;
import dev.robocode.tankroyale.botapi.events.*;

/**
 * Minimal stub for scale tests — spin radar, move a bit, fire on scan.
 */
public class StubBot extends Bot {
    public static void main(String[] args) {
        new StubBot().start();
    }

    @Override
    public void run() {
        while (isRunning()) {
            turnRadarRight(360);
            forward(40);
            turnLeft(15);
        }
    }

    @Override
    public void onScannedBot(ScannedBotEvent e) {
        fire(1.0);
    }

    @Override
    public void onHitWall(HitWallEvent e) {
        back(60);
        turnRight(90);
    }
}
